src/main/java/core/basesyntax/service/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.ReportGenerator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    @Override
+    public List<String> generateReport() {
+        Map<String, Fruit> allFruits = Storage.getAllFruits();
+        return allFruits.values().stream()
```

Comment:
The use of 'getAllFruits' directly from the 'Storage' class can be problematic in terms of encapsulation. It would be better to use a service layer to interact with the storage to ensure that any business logic related to data retrieval is encapsulated within the service.

Reply:
The logic was refactored